### ISSUE TITLE
Support loading of NPC data from the original Sentry plugin.

### ISSUE THREAD

**Issue by k-jiang** [Jabelpeeps/Sentries#3]
> [Citizens] The trait sentry failed to load for NPC ID: 4.
> [Citizens] The trait sentry failed to load for NPC ID: 6.
> [Citizens] The trait sentry failed to load for NPC ID: 14.
> [Citizens] The trait sentry failed to load for NPC ID: 21.
> [Citizens] The trait sentry failed to load for NPC ID: 23.

Tested on sentries 2.0.3 build 72
Citizens 2.0.19 build1381
Spigot 1.10.2 build 904

I used to have Spigot 1.9.4 and Sentry (1.8.8 build 36) and everything worked fine. This problem occurs after I update both my server and citizens to 1.10.2 compatible.

**Comment by Jabelpeeps** [comment]
Hi,

Thanks for trying Sentries!

I'm afraid there is no easy upgrade route from Sentry to my version.  I started my version in order to improve several aspects of the old codebase, two key ones being performance, and the command structure.

Unfortunately, in order to refactor the command and target handling as extensively as I have, I had to let go of backwards compatibility.  This was one of the reasons I renamed my version to Sentries - how did you find Sentries by-the-way?  I should probably add a note about this to the readme file...

I guess it would be theoretically possible to write an importer, though I'd want to finish other aspects before starting on that - I'll mark this as a feature request until then.

If you would like to give my version a go, I'd recommend you make a copy of the 'saves.yml' file in the citizens folder before reconfiguring your npc's to use Sentries.  It should then be easy to revert your npc's to their previous state if you decide it's not for you - just copy the backup copy back to the original location (when the server is not running).

As you can probably see from my commit history, I'm active most days with this and my other projects, so I'll (hopefully) be able to fix any bugs you discover quickly.  

Out of interest, are you running any of the other plugins that Sentry interfaces with?  (Towny, Factions, War, SimpleClans or Denizen?)  I went to some effort to keep the ones that were supported by the original codebase, but have tried to make it easier to add in other plugins in future.

Let me know if I can help in any other way.

**Comment by k-jiang** [comment]
Hi,

First of all, thank you for your great work on continuing the Sentry project. I was looking for a update to my old sentry plugin however the original project was abandoned. So I looked for different branches of the project and that's how I found this repo.

I'm a server op managing a free minecraft network in China (and I'm a chinese also so sorry for my bad english), so I don't know too much about programming stuff. But I can see it is not a easy challenge to reprogram the plugin and even make it faster and better than the previous one.

I'm willing to help on testing your plugin, but sadly we only have Citizens, Sentry and dtlTrader(the old one) install in our network so cannot help you with the other like Towny or Factions. The dtlTrader is in v3.3pre1 since the newer one require an extra plugin called "dtlCore" to be installed which adds some necessary load to our network. (http://dev.bukkit.org/bukkit-plugins/dtltraders/files/67-3-3-pre1/)

By the way I do use another plugin called Betonquest (http://betonquest.betoncraft.pl/) to program my Citizens NPC and write some quests and transcripts, yet it seems like the plugin is not integrated with sentry yet (or I don't know maybe?). Description and document are in the link if you want to have a look.

So far I forked and changed the original Sentry code a little bit  so that I can run our network safely before a newer upgrade is ready. Let me know if you have any updates so I can test it for you.

Personally speaking, you should publish the plugin on Spigotmc.org so ppl can know what happened (don't use Bukkit, it sucks, imo). And you will get more user who can support you or even give you donations. Also there's a official-like continuing of the sentry on here: https://github.com/mcmonkey4eva/Sentinel . Just saying if you think the project is too big for only a few people, then maybe collaborate with some other guys is never a bad idea.

**Comment by Jabelpeeps** [comment]
I plan to do a release on spigotmc once I have a few more bugs ironed out. I already have another plugin release on there - [BedHomes](https://www.spigotmc.org/resources/bedhomes.24270/)

Yeh, mcmonkey and Sentinel, I know about that project, if you are interested you could probably find the record of the conversation that he and I had about Sentry; basically he has gone his way, and I have gone mine.

Your english is fine, far better than my Chinese =)

If you would like to do the testing, I can look into setting up a way to integrate with BetonQuest;  what kind of integration would you be looking for?

**Comment by mcmonkey4eva** [comment]
@k-jiang Just a heads up, Sentinel is the modern official Sentry replacement, not this thing. Sentinel supports Sentry importing. https://www.spigotmc.org/resources/sentinel.22017/

(He spammed my repo, I'll spam his :P )

**Comment by k-jiang** [comment]
@mcmonkey4eva @Jabelpeeps Just curious, what's the big gap between you guys? Why go separately (and spamming me)? I could tell both of you are doing great jobs on maintaining your own repo. But there is no meaning on doing such a waste-of-energy thing. It is not going to work if you want to gain customers. Just several hours before I was looking at the Sentinel repo and was about to give it a test. But now I getting annoyed. I'm already tired of being bombed with viral marketing and advertisements so just please, please and please let me go!

And for the integration of BetonQuest (sorry for the late reply cause I was so busy and missed the notification), I think you guys can properly using the API which provided by BQ https://github.com/Co0sh/BetonQuest/wiki/Info-for-developers . Some basic usages would like, for example:
1. a player can hire/dismiss an NPC as a safeguard, or
2. the altitude of some Sentry/Sentries/Sentinel -- whatever you call it -- NPCs could be changed if conversations between players and a specific NPC goes wrong, or
3. the gears of the NPC could be changed based on BQ scripts, or
4. even the level, HP, item drops, spawn location, etc of an NPC can be changed too.
Those ideas just come from my unprofessional thinking so there may be some other more brilliant things to do so.

Although BetonQuest is perhaps the most easiest and powerful scripting plugin I ever meet (comparing to Denizen), yet it is kind of complicated. It took me few days to figure out how it works and be able to write some scripts on it. So take your time while studying on it :p . Also, the author of BQ is a very nice guy, try to contact him when you need some help.

**Comment by Jabelpeeps** [comment]
@k-jiang I do apologise for mcMonkey's behaviour.  He seems to be under the impression that his plugin is better than mine because he wrote it - at least I've seen no other information from him to support his position.

Regarding your feature request to work with BetonQuest, all of the tasks you have listed can be managed via the command structure, so if BetonQuest has the ability to run console commands, then you should be able to do what you want to your Sentries.  

One area that would be harder to manage via commands would be informing BetonQuest if a Sentries NPC had been attacked, or killed by a player, and I could look into the possibility of adding this if you are interested.

**Comment by mcmonkey4eva** [comment]
@k-jiang Sorry for the trouble, @Jabelpeeps seems to be under the impression that his plugin is better than the OFFICIAL CITIZENS TEAM CREATED SENTINEL PLUGIN, because ... nothing at all. He doesn't even have a reason as far as I can tell. The Citizens team has a long history of working together on things, (See: The Sentry original development, passed from fullwall to aufdemrand to jrbudda to myself over time), and jabel's rude forceful fight against us is highly disruptive to our attempt to provide a stable atmosphere.

I note that the only reason I posted here is because @Jabelpeeps thought it would be fun to spam my repo, so I, having nothing better to do, stooped to his level :/

**Comment by Jabelpeeps** [comment]
@k-jiang I have now implemented an importer for old Sentry NPC's into Sentries. 

If you want to give it a try, it is available in the current version on the Jenkins server (build #93 and above).

Please let me know in a new issue if you run into any problems with it.

I am closing this issue thread now, hopefully mcmonkey will stay away now.
